src/server/system_services/bucket_server.js

```diff
@@ -512,12 +514,39 @@ async function get_bucket_policy(req) {
     };
 }
 
+/** 
+    validate ARN principal
+    1. validate basic ARN, like arn prefix `arn:aws:iam::`
+    2. If principal ARN ends with `root` suffix, it's an account and get account with id eg: arn:aws:iam::${account_id}:root
++   3. if principal ARN contains `user`, it's an IAM user and get account with username and id
++        eg: arn:aws:iam::${account_id}:user/${iam_path}/${user_name}
+         account email = ${iam_user_name}:${account_id}
+
+    @param {SensitiveString | String} principal Bucket policy principal
+*/
+async function principal_validation_handler(principal) {
```

Comment:
This function is used in the parameter `get_account_handler` suggested to rename it to match its use, for example, `get_account_by principal`.